test/descriptor/pubspec.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async' show Future;
+import 'dart:convert' show utf8;
+import 'dart:io';
+
+import 'package:collection/collection.dart';
+import 'package:path/path.dart' as p;
+import 'package:test/test.dart';
+import 'package:test_descriptor/test_descriptor.dart';
+import 'package:yaml_edit/yaml_edit.dart';
+
+import '../descriptor.dart';
+
+class PubspecDescriptor extends FileDescriptor {
+  final String _contents;
+
+  PubspecDescriptor(String name, this._contents) : super.protected(name);
```

Comment:
Please leave comment revealing what format the `_contents` should be in.. I almost expect that this YAML :)

But documenting how it'll be compared it certainly nice.